django_app/redbox_app/redbox_core/models.py

```diff
@@ -51,22 +51,54 @@ def sanitise_string(string: str | None) -> str | None:
     return string.replace("\x00", "\ufffd") if string else string
 
 
+class ChatLLMBackend(models.Model):
+    """https://python.langchain.com/docs/how_to/chat_models_universal_init/"""
+
+    class Providers(models.TextChoices):
+        OPENAI = "openai"
+        ANTHROPIC = "anthropic"
+        AZURE_OPENAI = "azure_openai"
+        GOOGLE_VERTEXAI = "google_vertexai"
+        GOOGLE_GENAI = "google_genai"
+        BEDROCK = "bedrock"
+        BEDROCK_CONVERSE = "bedrock_converse"
+        COHERE = "cohere"
+        FIREWORKS = "fireworks"
+        TOGETHER = "together"
+        MISTRALAI = "mistralai"
+        HUGGINGFACE = "huggingface"
+        GROQ = "groq"
+        OLLAMA = "ollama"
+
+    name = models.CharField(max_length=128, help_text="The name of the model, e.g. “gpt-4o”, “claude-3-opus-20240229”.")
+    provider = models.CharField(max_length=128, choices=Providers, help_text="The model provider")
+    description = models.TextField(null=True, blank=True, help_text="brief description of the model")
+    is_default = models.BooleanField(default=False, help_text="is this the default llm to use.")
```

Comment:
should we add a enabled field so we can enable and disable models as required? Then update the query to only enclude enabled models?

Reply:
yes, good idea

Reply:
[3c15620](https://github.com/i-dot-ai/redbox/pull/1084/commits/3c15620267e3522d3b42e40b58b2172a25883299)